docs/design/adminconsole/INSTANCE-DATA.md

```diff
@@ -0,0 +1,280 @@
+  | odsInstanceDerivative.derivativeType | Max length: 50. Allowed values: "ReadReplica", "Snapshot". |
+
+* **Sample Payload**:
+
+  ```json
+  {
+    "odsInstanceId": 1,
+    "tenantId": 1,
+    "name": "Instance #1 - 2024",
+    "instanceType": "enterprise",
+    "odsInstanceContexts": [
+      {
+        "contextKey": "schoolYearFromRoute",
+        "contextValue": "2024"
+      }
+    ],
+    "odsInstanceDerivatives": [
+      {
+        "derivativeType": "ReadReplica"
+      }
+    ]
+  }
+  ```
+
+* **Response Format**:
+
+  ```json
+  {
+    "instanceId": 1,
+  }
+  ```
+
+### PUT /adminconsole/odsInstances/{id}
+
+* **Purpose**: Update an instance definition.
+* **Description**:
+  * Validate the incoming payload.
+  * Updates both the `adminconsole.Instance` and the `dbo.OdsInstances` tables.
+  * Respond with `202 No Content`
+* **Validation**:
+
+  | Property                             | Rules                                                      |
+  | ------------------------------------ | ---------------------------------------------------------- |
+  | name                                 | Max length: 100. Must be unique for the tenant.            |
+  | instanceType                         | Max length: 100.                                           |
+  | odsInstanceContexts                  | Empty array is allowed                                     |
+  | odsInstanceContext.contextKey        | Max length: 50. Must be unique for the instance.           |
+  | odsInstanceDerivatives               | Empty array is allowed                                     |
+  | odsInstanceContext.contextValue      | Max length: 50.                                            |
+  | odsInstanceDerivative.derivativeType | Max length: 50. Allowed values: "ReadReplica", "Snapshot". |
+
+* **Sample Payload**:
+
+  ```json
+  {
+    "odsInstanceId": 1,
+    "tenantId": 1,
+    "name": "Instance #1 - 2024",
+    "instanceType": "enterprise",
+    "odsInstanceContexts": [
+      {
+        "contextKey": "schoolYearFromRoute",
+        "contextValue": "2024"
+      }
+    ],
+    "odsInstanceDerivatives": [
+      {
+        "derivativeType": "ReadReplica"
+      }
+    ]
+  }
+  ```
+
+### DELETE /adminconsole/odsInstances/{id}
+
+* Not supported at this time. Respond with `405 Method Not Allowed`.
+
+### GET /adminconosle/instances
+
+Also supports `GET /adminconsole/instances/{id}`
+
+* **Purpose**: Provide instance list for the worker applications.
+* **Description**:
+  * Reads from the `adminconsole.Instance` table.
+  * Must be authorized with an appropriate Role name in the token.
+  * Returns a separate object for each ODS Instance Context.
+  * The `resourceUrl` is constructed from the tenant's base URL plus instance
+    context information.
+  * Respond with 200
+* **Query String**:
+  * `?status=` to search by Status.
+* **Response Format**:
+
+  ```json
+  [
+    {
```

Comment:
We have a `name` column for the Instance name in this table.

Reply:
"name" is usually not a very good name for a column or property, so I recommend calling this `instanceName` instead. I see that I should add `instanceId` as well, and I should note that `odsInstanceId` will be null if status != "Completed".

Reply:
done